feat(editor): add `oxc.fmt.experimental` flag

Sysix opened the pull request:
Support the new server option `fmt.experimental` with an own VSCode option `oxc.fmt.experimental`.
This option can be set for each Workspace individually. 

Added a Test to check if changing the settings, will tell the server and then the file can be formatted.

camc314 commented:
### Merge activity

* **Sep 20, 6:05 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
feat(editor): add `oxc.fmt.experimental` flag (#13923)

Support the new server option `fmt.experimental` with an own VSCode option `oxc.fmt.experimental`.
This option can be set for each Workspace individually.

Added a Test to check if changing the settings, will tell the server and then the file can be formatted.
